### PR TITLE
PLT-724: Remove Validator, MintingPolicy and StakeValidator from ledger api

### PR DESCRIPTION
**Pull request by zliu41**
...along with the Hash types. These can be moved to plutus-apps if needed.

### CODE REVIEW

File: plutus-ledger-api/src/PlutusLedgerApi/V1.hs [hunk 1]
```diff
@@ -75,19 +75,7 @@ module PlutusLedgerApi.V1 (
...
-    , ValidatorHash (..)
-    , MintingPolicy (..)
-    , mkMintingPolicyScript
-    , unMintingPolicyScript
-    , MintingPolicyHash (..)
-    , StakeValidator (..)
-    , mkStakeValidatorScript
-    , unStakeValidatorScript
-    , StakeValidatorHash (..)
+    -- *** Newtypes and hash types
```
Comment (koslambrou) [inline plutus-ledger-api/src/PlutusLedgerApi/V1.hs:78]: Should the `ScriptHash` be reexported here for consistency?

**Comment by michaelpj** [comment]
We have an open PR to improve the haddock for these: https://github.com/input-output-hk/plutus/pull/4814

Let's wait for that to merge, so that the final state that appears in this repository is a good one when it gets moved.

**Review by koslambrou — APPROVED** [review]

**Review by bezirg — APPROVED** [review]

**Comment by kwxm** [comment]
(I just want to see what's happening on the machine.)

**Comment by zliu41** [comment]
@kwxm I ssh'ed into the machine yesterday and ran `nix shell` on plutus, and for some reason it stuck, so I think that's the problem.

**Comment by kwxm** [comment]
> @kwxm I ssh'ed into the machine yesterday and ran `nix shell` on plutus, and for some reason it stuck, so I think that's the problem.

`nix-shell` works for me now, but it looks like the benchmarking job's still failing.

**Comment by zliu41** [comment]
@michaelpj can you merge it? The CI failure is unrelated.

**Comment by AriFordsham** [comment]
Do these types have a new home?

**Pull request merged**